path_config.go

```diff
@@ -0,0 +1,328 @@
+// Copyright © 2019, Oracle and/or its affiliates.
+package ociauth
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/vault/sdk/framework"
+	"github.com/hashicorp/vault/sdk/logical"
+	"strings"
+)
+
+const HOME_TENANCY_ID_CONFIG_NAME = "homeTenancyId"
+const BASE_CONFIG_PATH = "config/"
+
+var allowedConfigNamesForCreate = map[string]string{
+	HOME_TENANCY_ID_CONFIG_NAME: HOME_TENANCY_ID_CONFIG_NAME,
+}
+
+var allowedConfigNamesForUpdate = map[string]string{
+	HOME_TENANCY_ID_CONFIG_NAME: HOME_TENANCY_ID_CONFIG_NAME,
+}
+
+var allowedConfigNamesForDelete = map[string]string{
+	HOME_TENANCY_ID_CONFIG_NAME: HOME_TENANCY_ID_CONFIG_NAME,
+}
+
+func pathConfig(b *backend) *framework.Path {
```

Comment:
I'm confused as to the the structure of this config in a couple ways.

First, Vault backends generally have a single config and multiple roles. If you need another entire config, you mount a separate instance of the backend. Among other things, this allows you to manage the access to those separate instances via Vault's ACL system since they'll be mounted at different paths. But more importantly for this PR is just consistency... since I don't recall seeing another backend that allows multiple configs.

The other aspect is the data itself. I've tried to followed the use and as far as I can tell, the sole use of config data is during login to validate a tenant ID. Is that correct? If so I'd expect a single config object with tenant ID as the stored data.

Reply:
Hi, yes this config data is used only to validate the tenant Id. 

Is your suggestion to something like below?

`func (b *backend) pathConfig() *framework.Path {
	return &framework.Path{
		Pattern: "config/tenant$",
		Fields: map[string]*framework.FieldSchema{
			"home_tenantId": {
				Type:        framework.TypeString,
				Default:     "",
				Description: "Id of the home tenancy",
			},`

Reply:
Hi @mauruthig. I still have to review much of this PR, but in general an auth backend will have a single configuration that includes common setup items like namespace/tenant/project or whatever makes sense for that method, as well some credentials for connection, like an id/secret, or cert, etc. Creds are often search for in environment variables too, but usually it is possible to store this in the config. A good example of a simple config is the Azure Auth method: https://www.vaultproject.io/api/auth/azure/index.html#configure

Reply:
Thanks for your explanation Jim. The homeTenantId is only an additional safety check (its already enforced at OCI Identity Service). 

To simplify things, I am making a change to read the tenantId from an environment variable rather than an extensive config backend. Hence I will be remove the configuration related files.

Please let me know if you are ok with the above plan.

Reply:
Sounds OK.

Reply:
Hi Jim, 

I updated the config to be in line with other plugins in Vault. Can you take a look?